addons/base_import_module/models/ir_module.py

```diff
@@ -258,6 +268,125 @@ def module_uninstall(self):
             modules_to_delete.unlink()
         return res
 
+    @api.model
+    def web_search_read(self, domain, specification, offset=0, limit=None, order=None, count_limit=None):
+        res = super().web_search_read(domain, specification, offset=offset, limit=limit, order=order, count_limit=count_limit)
+        if any(dom for dom in domain if dom[0] == 'module_type' and dom[-1] != 'official'):
+            module_type = [dom for dom in domain if dom[0] == 'module_type'][0][-1]
+            fields = list(specification.keys())
+            modules_list = self._get_modules_from_apps(fields, module_type, False)
+            res['length'] += len(modules_list)
+            res['records'].extend(modules_list)
+        return res
+
+    def more_info(self):
+        return {
+            'name': _('Apps'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'ir.module.module',
+            'view_mode': 'form',
+            'res_id': self.id,
+            'context': self.env.context,
+        }
+
+    def web_read(self, specification):
+        fields = list(specification.keys())
+        module_type = self.env.context.get('module_type', 'official')
+        if module_type != 'official':
+            modules_list = self._get_modules_from_apps(fields, module_type, self.env.context.get('module_name'))
+            return modules_list
+        else:
+            return super().web_read(specification)
+
+    @api.model
+    def _get_modules_from_apps(self, fields, module_type, module_name):
+        payload = {
+            'series': major_version,
+            'module_fields': fields,
+            'module_type':module_type,
+            'module_name': module_name
+        }
+        headers = {'Content-Type': 'application/json'}
+        resp = requests.post(
+            f"{APPS_URL}/loempia/listdatamodules",
+            data=dumps({'params': payload}),
+            headers=headers,
+        )
+        modules_list = resp.json().get('result', [])
+        for mod in modules_list:
+            mod['id'] = -1
+            module_name = mod.get('name', module_name)
+            if 'icon' in fields:
+                mod['icon'] = f"{APPS_URL}{mod['icon']}"
+            if 'state' in fields:
+                existing_mod = self.search([('name', '=', module_name), ('state', '=', 'installed')])
+                if existing_mod:
+                    mod['state'] = 'installed'
+                else:
+                    mod['state'] = 'uninstalled'
+            if 'module_type' in fields:
+                mod['module_type'] = module_type
+            if 'website' in fields:
+                mod['website'] = f"{APPS_URL}/apps/modules/{major_version}/{module_name}/"
+            if 'app_iframe' in fields:
+                mod['app_iframe'] = '<iframe src="%s/?no_header=1" title="Apps Page"></iframe>' % mod['website']
```

Comment:
we want to display the website of the app vendor in an iframe? isn't that dangerous without huge benefit? maybe a max width and height?